### Check and Preparation of the data for training: 

Check if the data was inserted correclty.

Then, the data has to be filtered to Delft

In [ ]:
#Check size of data set, projection type, column titles
import geopandas as gpd

gdf = gpd.read_file("../data/bron_2022_2024.geojson")
print(gdf.shape)
print(gdf.crs)
print(gdf.columns.tolist())


(382421, 40)
EPSG:28992
['id', 'verkeersongeval_nummer', 'jaar_ongeval', 'verkeersongeval_afloop', 'aantal_partijen', 'partij_1_objecttype', 'partij_1_objecttype_overig', 'partij_2_objecttype', 'partij_2_objecttype_overig', 'aard_ongeval', 'niveau_koppelen', 'wegsituatie', 'bebouwde_kom', 'maximum_snelheid', 'wegverlichting', 'wegverharding', 'wegdek', 'lichtgesteldheid', 'zichtafstand', 'weersgesteldheid', 'bijz_verkeersmaatregel', 'bijz_verkeersmaatregel_overig', 'bijz_infrastructuur', 'bijz_infrastructuur_overig', 'bijz_tijdelijke_aard', 'bijz_tijdelijke_aard_overig', 'junctie_id', 'wegvak_id', 'hectometer', 'straatnaam', 'woonplaats', 'actueel', 'wegbeheerder', 'gemeente', 'provincie', 'dienstnaam', 'districtnaam', 'indicatie_alcohol', 'gdb_geomattr_data', 'geometry']


In [5]:
#Check for cyclist columns (= Fiets)
#Also integrate E-Bikes later?

print(gdf['partij_1_objecttype'].value_counts())

partij_1_objecttype
Personenauto                           233740
Bestelauto                              26899
Fiets                                   19599
Bromfiets                               18696
Snorfiets                                7328
Motor                                    6727
Trekker                                  3269
Vrachtauto                               2926
Bus                                      1756
e-bike                                   1688
Scootmobiel                               772
Overig wegmeubilair                       550
Brommobiel                                543
Landbouwvoertuig                          495
Lichtmast                                 301
Trein/tram                                208
Trekker met oplegger                      137
Boom                                      134
Voetganger                                112
Onbekend voertuig i.g.v. doorrijder        98
Dier                                       71
Overig vast ob

To filter out for Delft we project the data from EPSG:28992 (meters) to ESPG:4326 (lat/lon degrees) so that it is nicer to check the coordinates for Delft and to keep it consistent throughout the project.

In [3]:
#project data to lat/lon
gdf = gdf.to_crs("EPSG:4326")
print(gdf.crs)  # test if right format (EPSG:4326)

EPSG:4326


Now we want to filter for Delft:

(x-coordinates in [4.30,4.40] and y-coordinates in [51.95,52.05] in lat/lon)

In [ ]:
delft = gdf[
    (gdf.geometry.x >= 4.30) & (gdf.geometry.x <= 4.40) &
    (gdf.geometry.y >= 51.95) & (gdf.geometry.y <= 52.05) 
]
print(f"Accidents that happened in Delft: {len(delft)}")

Accidents that happened in Delft: 6025


Select the accidents involved with bikes (Fiets)

This results in 1116 total bike accidents in Delft

In [6]:
cyclists = delft[
    (delft['partij_1_objecttype'] == 'Fiets') |
    (delft['partij_2_objecttype'] == 'Fiets')
]
print(f"Cyclist accidents in Delft: {len(cyclists)}") 

Cyclist accidents in Delft: 1116


In [ ]:
#Save the filtered data in GeoJSON file to train on 
cyclists.to_file("../data/bron_delft_cyclists.geojson", driver="GeoJSON")